Logging/ARKLogMessage.h

```diff
@@ -27,15 +27,23 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARKLogMessage : NSObject <NSCopying, NSSecureCoding>
 
+/// @warning Do not use this initializer.
+/// @raises NSInternalInconsistencyException This is not a valid initializer.
+- (instancetype)init;
+
+/// Creates an ARKLogMessage with the provided parameters, created at the current date.
+/// @see initWithText:image:type:userInfo:creationDate:
 - (instancetype)initWithText:(NSString *)text image:(nullable UIImage *)image type:(ARKLogType)type userInfo:(nullable NSDictionary *)userInfo;
 
+- (instancetype)initWithText:(NSString *)text image:(nullable UIImage *)image type:(ARKLogType)type userInfo:(nullable NSDictionary *)userInfo creationDate:(NSDate *)date NS_DESIGNATED_INITIALIZER;
+
 @property (nonatomic, copy, readonly) NSDate *creationDate;
 @property (nonatomic, copy, readonly) NSString *text;
-@property (nonatomic, readonly) UIImage *image;
+@property (nonatomic, readonly, nullable) UIImage *image;
```

Comment:
hot damn, good find.

I'd love to do a pass that gets rid of `NS_ASSUME_NONNULL_*`, which would eliminate this class of errors... but haven't found the time yet.